src/main/java/hudson/plugins/build_timeout/impl/DeadlineTimeOutStrategy.java

```diff
@@ -26,11 +26,11 @@
 
     public static final int MINIMUM_DEADLINE_TOLERANCE_IN_MINUTES = 1;
 
-    protected static final String DEADLINE_REGEXP = new String("[0-2]?[0-9]:[0-5][0-9](:[0-5][0-9])?");
+    protected static final String DEADLINE_REGEXP = "[0-2]?[0-9]:[0-5][0-9](:[0-5][0-9])?";
 
-    protected static final SimpleDateFormat TIME_LONG_FORMAT = new SimpleDateFormat("H:mm:ss");
-    protected static final SimpleDateFormat TIME_SHORT_FORMAT = new SimpleDateFormat("H:mm");
-    protected static final SimpleDateFormat TIMESTAMP_FORMAT = new SimpleDateFormat("yyyy-MM-dd H:mm:ss");
+    protected static final String TIME_LONG_FORMAT_PATTERN = "H:mm:ss";
+    protected static final String TIME_SHORT_FORMAT_PATTERN = "H:mm";
+    protected static final String TIMESTAMP_FORMAT_PATTERN = "yyyy-MM-dd H:mm:ss";
```

Comment:
DeadlineTimeOutStrategy.java:-1, STCAL_STATIC_SIMPLE_DATE_FORMAT_INSTANCE, Priority: Normal

hudson.plugins.build_timeout.impl.DeadlineTimeOutStrategy.TIMESTAMP_FORMAT is a static field of type java.text.DateFormat, which isn't thread safe

As the JavaDoc states, DateFormats are inherently unsafe for multithreaded use. Sharing a single instance across thread boundaries without proper synchronization will result in erratic behavior of the application.

You may also experience serialization problems.

Using an instance field is recommended.

For more information on this see Sun Bug #6231579 and Sun Bug #6178997.